baaas-decision-fleet-manager/src/main/java/org/kie/baaas/dfm/app/manager/kafkaservice/ManagedKafkaService.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.kie.baaas.dfm.app.manager.kafkaservice;
+
+import javax.inject.Inject;
+
+import org.kie.baaas.dfm.app.managedservices.ManagedServicesClient;
+import org.kie.baaas.dfm.app.manager.KafkaService;
+import org.kie.baaas.dfm.app.model.eventing.Credential;
+import org.kie.baaas.dfm.app.vault.Secret;
+import org.kie.baaas.dfm.app.vault.VaultService;
+
+import static org.kie.baaas.dfm.app.managedservices.ManagedServicesClient.CLIENT_ID;
+import static org.kie.baaas.dfm.app.managedservices.ManagedServicesClient.CLIENT_SECRET;
+
+public class ManagedKafkaService implements KafkaService {
```

Comment:
The instance of this class is not being created by the IOC container, so I'm not sure the `@Inject` annotated fields will work here.

You will need to update the producer/factory method to have the dependencies injected and them pass them into this class as constructor params e.g:

```
public class KafkaServiceProducer {
  
   @Inject
   private ManagedServicesClient msc;
  
   @Produces
   public KafkaService produceKafkaService() {
         return new ManagedKafkaService(msc);
   }
}
```

Reply:
yes, agreed.